src/Apps/Artwork/artworkRoutes.tsx

```diff
@@ -21,10 +24,38 @@ export const artworkRoutes: AppRouteConfig[] = [
         artworkID,
       }
     },
+    render: ({ Component, props }) => {
+      if (!(Component && props)) {
+        return undefined
+      }
+
+      const requestError = (props as any).artworkResult?.requestError
+      const requestErrorStatusCode = requestError?.statusCode
+
+      const featureFlags = getENV("FEATURE_FLAGS")
+      const enableCustomErrorPage =
+        featureFlags?.["onyx_custom_artwork_error_page"]
+
+      if (requestErrorStatusCode) {
+        if (enableCustomErrorPage) {
+          updateContext("statusCode", requestErrorStatusCode)
+        } else {
+          throw new HttpError(requestErrorStatusCode)
+        }
+      }
+
+      return <Component {...props} />
+    },
     query: graphql`
       query artworkRoutes_ArtworkQuery($artworkID: String!) {
-        artwork(id: $artworkID) @principalField {
-          ...ArtworkApp_artwork
+        artworkResult(id: $artworkID) {
```

Comment:
**question (non-blocking):** If `artworkResult` covers all possible outcomes, success or error, does it make sense keep it as a `@principalField`?

Reply:
Exactly, the `artworkResult` style field supercedes/is no longer appropriate to use with `@principalField`.

The directive was a way to elevate errors in a first-class way to `extensions`, adding a little object shape to the generally schema-less and free-form `extensions`. This was preferable to `errors`, where it's harder to find + parse, for a client.

However with `artworkResult` (introduced in https://github.com/artsy/metaphysics/pull/5537), we're no longer throwing any error associated with loading the artwork at all (nothing in `errors` or `extensions`), and any error encountered is elevated to a first-class object in the returned schema via that union type. That's actually even better than the `@principalField` directive.

So, keeping the directive here wouldn't really have any effect as any errors the directive ordinarily would cover are no longer being thrown.